src/JSPropUpdater.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  NativeEventEmitter,
+  NativeModules,
+  findNodeHandle,
+} from 'react-native';
+import { nativeShouldBeMock } from './reanimated2/PlatformChecker';
+import { StyleProps } from './reanimated2';
+
+interface ListenerData {
+  viewTag: number;
+  props: StyleProps;
+}
+
+export class JSPropUpdater {
+  private static _tagToComponentMapping = new Map();
+  private _reanimatedEventEmitter: NativeEventEmitter;
+  private static _reanimatedModuleMock = {
+    async addListener(): Promise<void> {
+      // noop
+    },
+    async removeListeners(): Promise<void> {
+      // noop
+    },
+  };
+
+  private static _listener(data: ListenerData) {
+    const component = JSPropUpdater._tagToComponentMapping.get(data.viewTag);
+    component && component._updateFromNative(data.props);
```

Comment:
I know this snippet of code was copied but it may be finally worth to convert it into a proper conditional.